src/base/pegasus_value_schema.h

```diff
@@ -8,105 +8,106 @@
 #include <string.h>
 #include <string>
 #include <vector>
+
 #include <dsn/utility/ports.h>
 #include <dsn/utility/utils.h>
-#include <dsn/utility/blob.h>
+#include <dsn/utility/smart_pointers.h>
+#include <dsn/utility/endians.h>
 #include <dsn/service_api_c.h>
 #include <rocksdb/slice.h>
 
 namespace pegasus {
 
 #define PEGASUS_VALUE_SCHEMA_MAX_VERSION 0
 
-// =====================================================================================
-// rocksdb value (version 1) = [expire_ts(uint32_t)] [user_data(bytes)]
-
-// generate rocksdb value
-// T may be std::string or ::dsn::blob or rocksdb::slice.
-// 'user_data' will not be copied, so 'user_data' may be referenced by 'slices'.
-// some data may be put in 'buf', so 'buf' may be referenced by 'slices'.
-template <typename T>
-void pegasus_generate_value(uint32_t version,
-                            uint32_t expire_ts,
-                            T &user_data,
-                            std::string &buf,
-                            std::vector<rocksdb::Slice> &slices)
+/// Extracts expire_ts from rocksdb value with given version.
+/// The value schema must be in v0.
+/// \return expire_ts in host endian
+inline uint32_t pegasus_extract_expire_ts(int version, dsn::string_view value)
```

Comment:
abseil 官方推荐是不需要传 const 引用

```
// Because of its small size, prefer passing `string_view` by value:
//
//   void MyFunction(dsn::string_view arg);
//
// If circumstances require, you may also pass one by const reference:
//
//   void MyFunction(const dsn::string_view& arg);  // not preferred
//
// Passing by value generates slightly smaller code for many architectures.
```
这是 dsn/utility/string_view.h 下的注释.